File: test/StringThing.SqlClient.Dapper.IntegrationTests/SqlServerFixture.cs
using Testcontainers.MsSql;
using Xunit;

namespace StringThing.SqlClient.Dapper.IntegrationTests;

public class SqlServerFixture : IAsyncLifetime
{
    private readonly MsSqlContainer _container = new MsSqlBuilder("mcr.microsoft.com/mssql/server:2022-latest")
        .Build();

    public string ConnectionString { get; private set; } = null!;

    public async ValueTask InitializeAsync()
    {
        await _container.StartAsync();
        ConnectionString = _container.GetConnectionString();
    }

    public async ValueTask DisposeAsync()
    {
        await _container.DisposeAsync();
    }
}

File: fsharp/test/StringThing.FSharp.SqlClient.IntegrationTests/SqlServerFixture.fs
namespace StringThing.FSharp.SqlClient.IntegrationTests

open System.Threading.Tasks
open Testcontainers.MsSql
open Xunit

/// xUnit class fixture that boots a SQL Server (mssql) container once for the test class
/// and tears it down at the end.
type SqlServerFixture() =
    let container =
        MsSqlBuilder()
            .WithImage("mcr.microsoft.com/mssql/server:2022-latest")
            .Build()
    let mutable connectionString = ""

    member _.ConnectionString = connectionString

    interface IAsyncLifetime with
        member _.InitializeAsync() : ValueTask =
            ValueTask(
                task {
                    do! container.StartAsync()
                    connectionString <- container.GetConnectionString()
                } :> Task)

        member _.DisposeAsync() : ValueTask =
            container.DisposeAsync()
